### PR TITLE
Vec re-implementation

### PR DESCRIPTION
**Pull request by luc-tielen**
- [x] Mark this if you consider it ready to merge
- [x] I've added tests (optional)
- [ ] I wrote some lines in the [book](https://github.com/radareorg/radare2book) (optional)

**Description**

### CODE REVIEW

**Comment by trufae** [comment]
Looks good!

**Review by trufae — CHANGES_REQUESTED** [review]
those comments are probably deprecated so feel free to ignore them

File: shlr/sdb/include/sdb/r_vec.h [hunk 1]
```diff
@@ -0,0 +1,344 @@
+#ifndef R2_VEC_H
+#define R2_VEC_H
+
+#include <r_util/r_assert.h>
+#include <r_types.h>
+#include <stdlib.h>
```
Comment (trufae) [inline shlr/sdb/include/sdb/r_vec.h:6]: i think stdlib and stdbool are already included in r_types.

File: libr/include/r_vec.h [hunk 2]
```diff
@@ -0,0 +1,224 @@
...
+	} \
+	static inline void R_VEC_FUNC(type, shrink_to_fit)(R_VEC(type) *vec) { \
+		r_return_if_fail (vec); \
+		const size_t num_elems = R_VEC_FUNC(type, length) (vec); \
+		if (num_elems == R_VEC_FUNC(type, capacity) (vec)) { \
+			return; \
+		} \
+		type *buf = realloc (vec->start, num_elems * sizeof (type)); \
+		if (R_UNLIKELY (!buf)) { \
+			return; \
```
[the remark this answers is not in the document]
Comment (luc-tielen) [reply libr/include/r_vec.h]: I thought about this but I think this check is better here, realloc could sometimes fail if we hit OOM error.. but probably we have bigger problems then.

File: shlr/sdb/include/sdb/r_vec.h (hunk 1 above)
Comment (luc-tielen) [reply shlr/sdb/include/sdb/r_vec.h:6]: Ah nice.

**Pull request merged**